### DISCUSSION TITLE
[Urgent] @walletconnect/ethereum-provider `provider.connect()` Unable to invoke the wallet app in an iframe.

### DISCUSSION THREAD

**Discussion by liubin595338764** [WalletConnect/walletconnect-monorepo#3380] [Developer Support]
**Describe the bug**
@walletconnect/ethereum-provider `provider.connect()` Unable to invoke the wallet app in an iframe.

In Safari on iPhone.

Staying on this interface without any response.
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/5274428/fdc689ea-f3c2-4f2a-af6e-0353c8abdab9)

**SDK Version (if relevant)**
 - "@walletconnect/ethereum-provider": "^2.8.6",

**Smartphone (please complete the following information):**
 - Device: iPhone11
 - OS: ios 15.4.1
 - Browser safari

**Comment by liubin595338764** [comment]
@glitch-txs  Could you please help me take a look at this issue? Thank you!

**Reply by glitch-txs** [reply]
hi! Could you solve this?

**Reply by liubin595338764** [reply]
@glitch-txs  I don't understand, I can't solve this problem.

**Comment by arein** [comment]
This needs better repro steps.
Do you have a test website where we can repro this?

**Reply by homeyong** [reply]
Hello @arein,

Etherscan here. 
We're currently facing a similar issue on our contract tab. It's constructed using an Iframe, and we're unable to initiate the wallet application too.

Could you kindly provide us with assistance regarding this matter? Here's an example of a contract where this issue is occurring:

Sample Contract: [Link](https://arbiscan.io/address/0x1c66cdee4f63f703fc64776227e0bd576d118e56#writeContract)

Thank you!

**Reply by homeyong** [reply]
This functionality operates smoothly on Android devices and functions as expected on Chrome for iOS. 
It only encounters issues specifically on Safari for iOS.

**Reply by alipouryousefi** [reply]
anyone solve this?
